### PR TITLE
Kusto 2020-06-14 PS version

### PR DESCRIPTION
**Pull request by astauben**
## Description

Design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/685

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Kusto/.gitignore [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+internal
+exports
+custom/*.psm1
+test/*-TestResults.xml
+/*.ps1
+/*.ps1xml
+/*.psm1
+/*.snk
+/*.csproj
+/*.nuspec
```
Comment (isra-fel) [inline src/Kusto/.gitignore:14]: Please ignore `tools` folder as well and don't check in any files in it

File: src/Kusto/test/Add-AzKustoDatabasePrincipal.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+$currentPath = $PSScriptRoot
+while(-not $mockingPath) {
+    $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+    $currentPath = Split-Path -Path $currentPath -Parent
+}
+. ($mockingPath | Select-Object -First 1).FullName
+
+Describe 'Add-AzKustoDatabasePrincipal' {
+    It 'AddExpanded' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (isra-fel) [inline src/Kusto/test/Add-AzKustoDatabasePrincipal.Tests.ps1:16]: Why are test cases all skipped?

Comment (astauben) [reply src/Kusto/test/Add-AzKustoDatabasePrincipal.Tests.ps1:16]: we are testing the scenario in the New-AzKustoDatabasePrincipalAssignment file.

File: src/Kusto/.gitignore (hunk 1 above)
Comment (astauben) [reply src/Kusto/.gitignore]: done

**Comment by astauben** [comment]
> Hi @astauben please see my inline comments.
> Besides, it would be great if you could update the examples with the new parameters.

Hi, @isra-fel Thanks fow the review. all comments are resolved

File: src/Kusto/.gitignore (hunk 1 above)
Comment (isra-fel) [reply src/Kusto/.gitignore]: Some of the files are already checked in, e.g. `src/Kusto/tools/Resources/examples/readme.md`. Please remove the whole tools/ folder and commit again. Thanks

File: src/Kusto/test/Add-AzKustoDatabasePrincipal.Tests.ps1 (hunk 2 above)
Comment (isra-fel) [reply src/Kusto/test/Add-AzKustoDatabasePrincipal.Tests.ps1:16]: Can you just delete this file?

Comment (astauben) [reply src/Kusto/test/Add-AzKustoDatabasePrincipal.Tests.ps1]: done

File: src/Kusto/.gitignore (hunk 1 above)
Comment (astauben) [reply src/Kusto/.gitignore]: done

**Pull request merged**